packages/api/src/routes/middlewares/auth.ts

```diff
@@ -9,49 +11,77 @@ import { getCxIdOrFail } from "../util";
  * Process the API key and get the customer id.
  * The customer id is stored on the Request, property 'cxId'.
  */
-export const processAPIKey = (req: Request, res: Response, next: NextFunction): void => {
+export function processCxId(req: Request, res: Response, next: NextFunction): void {
+  const { log } = out("processCxId");
   try {
     // Just gets the cxId from the API Key, the actual auth is done on API GW.
     // Downstream routes should check whether `cxId` is present on the request or not.
     const encodedApiKey = req.header("x-api-key");
     req.cxId = getCxIdFromApiKey(encodedApiKey);
+    // TODO 1935 remove this
+    // TODO 1935 remove this
+    // TODO 1935 remove this
+    log(`..... cxId from API Key: ${req.cxId}`);
   } catch (error) {
-    // noop - auth is done on API GW level, this is just to make data available downstream
+    try {
+      // TODO 1935 remove this
+      // TODO 1935 remove this
+      // TODO 1935 remove this
+      req.cxId = getCxIdFromJwt(req);
+      log(`..... cxId from JWT: ${req.cxId} 🤘`);
+    } catch (error) {
+      // noop - auth is done on API GW level, this is just to make data available downstream
+    }
```

Comment:
nit: Couldnt we do a if else so that if `encodedApiKey` is true use `getCxIdFromApiKey` else use `getCxIdFromJwt` instead of nested try catches?

Reply:
`getCxIdFromApiKey` has logic that might fail if the API Key is set but invalid.